zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/Follower.java

```diff
@@ -159,6 +159,22 @@ protected void processPacket(QuorumPacket qp) throws Exception {
         case Leader.PING:
             ping(qp);
             break;
+        case Leader.SKIP:
+            // It is sad that we have no breadcrumbs to route packet back to origin. Basically, we have choices:
+            // 1. Attach breadcrumbs to outstanding request.
+            //    This request protocol changes and protocol version bumping.
```

Comment:
I think we could also cover #2069 if we want to go this way. I prefer this actually. We will inevitably introduce protocol in long term future. If so, why now ? But it still could be a separate issue before 3.10.0.